### PR TITLE
BI-1612 - Experiment Import Phenotype Data Persistence

### PR DESCRIPTION
**Pull request by nickpalladino**
# Description
**Story:** [BI-1612](https://breedinginsight.atlassian.net/browse/BI-1612?atlOrigin=eyJpIjoiMzJhZTZmNDY4ZjRmNDdhZGFlMzcwODRjMzJiOTUzYjMiLCJwIjoiaiJ9)

- Added minimal BrAPIObservation construction for import which is persisted to the database and returned to the front end
- Added observations to import statistics

# Dependencies
- bi-web develop

# Testing
- Evaluate how blank observation values are handled and how we want that to work. Right now it just ignores them and will not include in the preview data or posted data.
- Change to PendingImport may affect prototype importer and should probably be regression tested at some point

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have tested that my code works with both the brapi-java-server and BreedBase
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<please include a link to TAF run>_

### CODE REVIEW

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java [hunk 1]
```diff
@@ -293,14 +317,32 @@ private void getNewBrapiData(List<BrAPIImport> importRows, Program program, User
...
+
+            for (Column<?> column : phenotypeCols) {
+                PendingImportObject<BrAPIObservation> obsPIO = createObservationPIO(importRow, column.name(), column.getString(i));
+                this.observationByHash.put(getImportObservationHash(importRow, getVariableNameFromColumn(column)), obsPIO);
+            }
         }
     }
 
-    private String createObservationUnitKey(ExperimentObservation importRow) {
+    private static String createObservationUnitKey(ExperimentObservation importRow) {
```
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java:328]: Why was this changed to be static?

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java [hunk 2]
```diff
@@ -293,14 +317,32 @@ private void getNewBrapiData(List<BrAPIImport> importRows, Program program, User
...
+    }
+
+    //TODO: Add timestamp parameter once we support them
+    private static String getObservationHash(String observationUnitName, String variableName, String studyName) {
+        String concat = DigestUtils.sha256Hex(observationUnitName) +
+                DigestUtils.sha256Hex(variableName) +
+                DigestUtils.sha256Hex(studyName);
+        return DigestUtils.sha256Hex(concat);
+    }
+
```
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java:345]: Why are these methods static?

**Review by timparsons — CHANGES_REQUESTED** [review]
I need to test the code still, but had two minor questions

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java (hunk 2 above)
Comment (nickpalladino) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java:345]: I had them as static in the old ObservationProcessor and just brought that over because there is no dependence on the class instance.

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java (hunk 1 above)
Comment (nickpalladino) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java:328]: Think I just changed it when I was in there in line with the rationale in the following comment.

Comment (nickpalladino) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java]: Pushed changes.

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java (hunk 2 above)
Comment (nickpalladino) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java]: Pushed changes.

**Review by timparsons — APPROVED** [review]
Testing passed.

**Review by dmeidlin — APPROVED** [review]
